### PR TITLE
[Draft] teams: init at 1.2.00.32451

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
wanted to add the native teams app to nixpkgs

it's able to build, but currently i get this when running it:

```
strace ./result/share/teams/teams
...
connect(87, {sa_family=AF_UNIX, sun_path="/run/user/1000/pulse/native"}, 110) = 0
sendto(85, "W", 1, MSG_NOSIGNAL, NULL, 0) = -1 ENOTSOCK (Socket operation on non-socket)
write(85, "W", 1)                       = 1
write(85, "W", 1)                       = 1
mmap(NULL, 8392704, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_STACK, -1, 0) = 0x7f2fb17fb000
mprotect(0x7f2fb17fc000, 8388608, PROT_READ|PROT_WRITE) = 0
clone(child_stack=0x7f2fb1ffaa30, flags=CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID, parent_tid=[32182], tls=0x7f2fb1ffb700, child_tidptr=0x7f2fb1ffb9d0) = 32182
futex(0x55ecc13cd580, FUTEX_UNLOCK_PI_PRIVATE) = 0
futex(0x55ecc14422a8, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
futex(0x55ecc13cd580, FUTEX_LOCK_PI_PRIVATE, NULL) = 0
futex(0x55ecc14422ac, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
futex(0x55ecc13cd580, FUTEX_LOCK_PI_PRIVATE, NULL) = 0
read(58, "", 4)                         = 0
close(58)                               = 0
futex(0x55ecc13cd580, FUTEX_UNLOCK_PI_PRIVATE) = 0
futex(0x55ecc14422a8, FUTEX_WAIT_PRIVATE, 0, NULL) = 0
futex(0x55ecc13cd580, FUTEX_UNLOCK_PI_PRIVATE) = 0
gettid()                                = 32146
futex(0x7f2fb27fb7b8, FUTEX_WAKE_PRIVATE, 2147483647) = 1
futex(0x7f2fb27fb768, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x7f2f9c001c38, FUTEX_WAKE_PRIVATE, 1) = 1
read(41, "U\251m\377\34X\276\32", 8)    = 8
write(17, "\0", 1)                      = 1
write(17, "\0", 1)                      = 1
+++ killed by SIGTRAP (core dumped) +++
Trace/breakpoint trap (core dumped)
```
I'm just not really sure where to go from here, I'm assuming it's having some difficulty interacting with pulseaudio then dies, but I'm not really sure.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by jonringer** [comment]
@worldofpeace can you think of anything?

**Comment by Elyhaka** [comment]
@jonringer I have been trying to package teams this moring, and I ended up in the same situation. (Same place in the strace). I've later found #75505 and after fixing it (commented it there), it started to work. I'm not really sure why it works with this one, though.

**Comment by aanderse** [comment]
Looking forward to this

BTW @jonringer what do you use for O365 (specifically mail, though interested in others too), assuming you run NixOS as your daily driver.

**Comment by JohnAZoidberg** [comment]
> what do you use for O365 (specifically mail, though interested in others too)

outlook.office.com is pretty good

File: pkgs/applications/networking/instant-messengers/teams/default.nix [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    #mv * $out/opt/${binaryName}
+
+    #chmod +x $out/opt/${binaryName}/${binaryName}
+    #patchelf --set-interpreter ${stdenv.cc.bintools.dynamicLinker} \
+    #    $out/opt/${binaryName}/${binaryName}
+
+    wrapProgram $out/share/${binaryName}/${binaryName} \
+        "''${gappsWrapperArgs[@]}" \
+        --prefix XDG_DATA_DIRS : "${gtk3}/share/gsettings-schemas/${gtk3.name}/" \
+
```
Comment (worldofpeace) [inline pkgs/applications/networking/instant-messengers/teams/default.nix]: if gtk3, glib, and wrapGAppsHook are present in the proper *inputs, this is completely unnecessary.
`wrapGAppsHook` will wrap automatically, and `glib` setup-hook will collect gtk3's gsettings-schemas directed passed to wrapGAppsHook.

**Comment by jonringer** [comment]
Yea, i use the O365 browser versions. But having a native one is nicer for responsiveness and better notifications

**Comment by aanderse** [comment]
> Yea, i use the O365 browser versions. But having a native one is nicer for responsiveness and better notifications

@jonringer rough. That is what I'm currently stuck on too :frowning_face:
@JohnAZoidberg we'll have to agree to disagree on this point, though mostly because of calendar notifications.

**Comment by worldofpeace** [comment]
I think we'll go with  #75505

**Pull request closed without merging**